core/templates/dev/head/i18n.js

```diff
@@ -43,7 +43,23 @@ oppia.controller('I18nFooter', [
   // Changes the language of the translations.
   var preferencesDataUrl = '/preferenceshandler/data';
   var siteLanguageUrl = '/save_site_language';
-  $scope.supportedSiteLanguages = GLOBALS.SUPPORTED_SITE_LANGUAGES;
+
+  var items = Object.keys(GLOBALS.SUPPORTED_SITE_LANGUAGES).map(function(key) {
+    return [key, GLOBALS.SUPPORTED_SITE_LANGUAGES[key]];
+  });
+
+  items.sort(function(first, second) {
+    var textA = first[1].toUpperCase();
+    var textB = second[1].toUpperCase();
+    return (textA < textB) ? -1 : (textA > textB) ? 1 : 0;
+  });
+
+  var supportedSiteLanguages = {};
```

Comment:
For clarity, define this as `$scope.supportedSiteLanguages = {}` and use `$scope.supportedSiteLanguages` to access the variable.